electron/src/window/WindowManager.ts

```diff
@@ -68,14 +69,25 @@ export class WindowManager {
     }
   }
 
+  static flushActionsQueue() {
+    const actions = WindowManager.actionsQueue;
+    if (actions) {
+      actions.forEach(({action, args}) => this.sendActionToPrimaryWindow(action, ...args));
+      WindowManager.actionsQueue = [];
+    }
+  }
+
   static async sendActionAndFocusWindow(action: string, ...args: any[]): Promise<void> {
     await app.whenReady();
 
     const primaryWindow = WindowManager.getPrimaryWindow();
 
     if (primaryWindow) {
       if (primaryWindow.webContents.isLoading()) {
-        primaryWindow.webContents.once('did-finish-load', () => primaryWindow.webContents.send(action, ...args));
+        primaryWindow.webContents.once('did-finish-load', () => {
+          WindowManager.actionsQueue = [...WindowManager.actionsQueue, {action, args}];
+          primaryWindow.webContents.send(action, ...args);
+        });
```

Comment:
We do not need to listen for the `did-finish-load` event anymore, do we? 

```suggestion
// If the webapp is not yet loaded we queue the action we want to send. It will be flushed later on by the flushActionsQueue` method
WindowManager.actionsQueue.push({action, args});
```